src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/WarningPropertiesCollection.cs

```diff
@@ -77,6 +77,17 @@ public bool ApplyNoWarnProperties(IRestoreLogMessage message)
             return ApplyProjectWideNoWarnProperties(message, ProjectWideWarningProperties) || ApplyPackageSpecificNoWarnProperties(message);
         }
 
+        /// <summary>
+        /// Attempts to suppress a warning log message.
+        /// The decision is made based on the Package Specific no warn properties.
+        /// </summary>
+        /// <param name="message">Message that should be suppressed.</param>
+        /// <returns>Bool indicating is the warning should be suppressed or not.</returns>
+        public bool ApplyNoWarnProperties(IPackLogMessage message)
```

Comment:
This and the other new method added to this class are pack specific, whereas the filename is under the `RestoreCommand` directory.

I think it would be better to extract the shared code into an internal abstract class, keep the restore-specific code here, and create a new pack specific class with these new methods.

Or, move the class to a new common/shared location, and then make this file extend that class, without any body, to keep API & ABI compatibility.

Reply:
Thank you for your recommendation.
I created below 2 new files.
`NuGet.Client\src\NuGet.Core\NuGet.Commands\PackCommand\Logging\PackageSpecificWarningProperties.cs`
`C:\NuGetProj\NuGet.Client\src\NuGet.Core\NuGet.Commands\PackCommand\Logging\WarningPropertiesCollection.cs`

Please check now.